### PR TITLE
CBL-4627: Update Deployment Target to iOS 11.0 and macOS 10.14

### PR DESCRIPTION
**Pull request by velicuvlad**
Besides the relevant project config change, the rest of updates are within tests or utility functions - so no affect on the product itself

### CODE REVIEW

File: CouchbaseLite.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6812,7 +6812,7 @@
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				INFOPLIST_FILE = Swift/Tests/Info.plist;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks @executable_path/../Frameworks @loader_path/../Frameworks";
-				MACOSX_DEPLOYMENT_TARGET = 10.12;
+				MACOSX_DEPLOYMENT_TARGET = 10.14;
```
Comment (pasin) [inline CouchbaseLite.xcodeproj/project.pbxproj:6815]: Don't do it here in project.pbxproj.

File: Objective-C/Internal/Replicator/CBLTrustCheck.mm [hunk 2]
```diff
@@ -113,7 +113,7 @@ - (BOOL) forceTrusted {
     SecTrustSetExceptions(_trust, exception);
     CFRelease(exception);
 
-    if (@available(iOS 12.0, macos 10.14, *)) {
+    if (@available(iOS 12.0, *)) {
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLTrustCheck.mm:116]: For minimum change in release branch, let do not change this.

File: Objective-C/Internal/Replicator/CBLTrustCheck.mm [hunk 3]
```diff
@@ -149,7 +149,7 @@ - (NSURLCredential*) checkTrust: (NSError**)outError {
     SecTrustResultType result;
     OSStatus err;
 
-    if (@available(iOS 12.0, macos 10.14, *)) {
+    if (@available(iOS 12.0, *)) {
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLTrustCheck.mm:152]: For minimum change in release branch, let do not change this.

File: Objective-C/Internal/Replicator/CBLWebSocket.mm [hunk 4]
```diff
@@ -271,16 +271,12 @@ - (void) setupAuth {
...
-                if (@available(macOS 10.12, iOS 10.0, *)) {
-                    _clientIdentity = toSecIdentityWithCertChain(c4cert, &error);
-                    if (_clientIdentity) {
-                        c4cert_release(c4cert);
-                        return;
-                    }
-                    CBLWarnError(Sync, @"%@: Couldn't lookup the identity from the KeyChain: %@", self, error);
-                } else {
-                    CBLWarnError(Sync, @"%@: Client Cert Auth is not supported by macOS < 10.12 and iOS < 10.0", self);
+                _clientIdentity = toSecIdentityWithCertChain(c4cert, &error);
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLWebSocket.mm:274]: For minimum change in release branch, let do not change this.

File: Objective-C/Tests/PredictiveQueryTest+CoreML.m [hunk 5]
```diff
@@ -19,7 +19,6 @@
 #import "CBLTestCase.h"
 #import "CBLCoreMLPredictiveModel+Internal.h"
 
-API_AVAILABLE(macos(10.13), ios(11.0))
 @interface PredictiveQueryWithCoreMLTest : CBLTestCase
```
Comment (pasin) [inline Objective-C/Tests/PredictiveQueryTest+CoreML.m:22]: For minimum change in release branch, let do not change this.

File: Objective-C/Tests/PredictiveQueryTest+CoreML.m [hunk 6]
```diff
@@ -303,7 +302,7 @@ - (void) testMultiArrayDataConversion {
 }
 
 - (void) testSequenceDataConversion {
-    if (@available(macOS 10.14, iOS 12.0, *)) {
+    if (@available(iOS 12.0, *)) {
```
Comment (pasin) [inline Objective-C/Tests/PredictiveQueryTest+CoreML.m:305]: For minimum change in release branch, let do not change this.

**Review by pasin — CHANGES_REQUESTED** [review]
For minimum change in release branch, please only change the version in xcconfig file.

File: Package.swift [hunk 7]
```diff
@@ -4,7 +4,7 @@ import PackageDescription
 let package = Package(
     name: "CouchbaseLiteSwift",
     platforms: [
-        .iOS(.v10), .macOS(.v10_12)
+        .iOS(.v11), .macOS(.v10_14)
```
Comment (pasin) [inline Package.swift:7]: We will do this change when releasing 3.1.1.

**Review by pasin — CHANGES_REQUESTED** [review]

**Comment by jianminzhao** [comment]
Looks good to me, but @pasin is the expert.

**Review by jianminzhao — APPROVED** [review]

**Review by pasin — APPROVED** [review]

**Pull request merged**